src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -314,6 +315,14 @@ private string FinishFixingName(string name)
                 words = name.Split(new[] { WordSeparator }, StringSplitOptions.RemoveEmptyEntries);
             }
 
+            if (CapitalizationScheme == Capitalization.AllLower && !string.IsNullOrEmpty(WordSeparator))
+            {
+                if (words.Count() == 1) // Only Split if words have not been split before 
+                {
+                    words = Regex.Split(name, @"(?<!^)(?=[A-Z])");
```

Comment:
Could we use `Microsoft.CodeAnalysis.Shared.Utilities.StringBreaker` instead of `Regex`

Reply:
I'm not familiar with that API as this will be the first issue I have worked on in Roslyn could you point me to the documentation and i will look into it

Reply:
Done